### PR TITLE
feat: [M13] Swift: interpretability extras — mixing_matrix, hidden_states accessor, verify_block

### LINKED ISSUE #264: [M13] Swift: interpretability extras — mixing_matrix, hidden_states accessor, verify_block
Follow-up from #166 (PR #262).

Not ported to `swift/engine/`:

- `SelectiveSSM.mixing_matrix` / `MambaBlock.mixing_matrix` (`src/model/mlx_backend.py:280-298`, `:426-435`) — the dense `(B,H,L,L)` 1-semiseparable mixing matrix
- `MLXMambaModel.mixing_matrices` (`:869-881`)
- **`verify_block` (`:828-852`) is the real one**: it is a prerequisite for #172 (speculative decoding), because the wall-clock win comes from evaluating a whole candidate block in ONE graph eval rather than paying per-token kernel-launch/sync.

`hidden_states` IS ported (`MonicaModel.hiddenStates`), because `monica-parity` needs it to localize a divergence to a single layer — but it is not exposed as a CLI surface.

### PR DESCRIPTION
**Pull request by travisgalloway**
Closes #264

## Summary
- Ports #100/#52's Python interpretability/speculative-decoding auxiliaries to `swift/engine/`: `SelectiveSSM.mixingMatrix`, `MambaBlock.mixingMatrix`/`ssmInput`, `MonicaModel.mixingMatrices`/`verifyBlock`, `LayerState.arrays`.
- Fixes `MLXMambaModel.mixing_matrices`'s MoE guard (attention-only — crashed on MoE configs; CUDA's guard already covered both); adds `tests/test_mlx_mixing_matrix.py` pinning the identity/causality/layer-skip properties before freezing the oracle into a fixture.
- `monica-parity` gains three strict fp32 gates (`rtol=1e-4/atol=1e-5`, no tolerance loosened anywhere): hidden-state comparison promoted from failure-only to unconditional; a mixing-matrix section (index-list equality, per-layer compare, in-process causality + defining-identity checks on the un-averaged tensor); a `verifyBlock` section (logits vs `step_logits`, final state vs `prefill.safetensors`, an in-process rollback identity).
- `monica-generate` gains `--dump-activations <file.safetensors>` for a Python-diffable Swift-side dump.
- Regenerates the six non-quantized fixtures with `mixing.{i}`/`mixing_layers`; `toy-moe-int8`/`toy-moe-int4` are untouched.
- Along the way, found and fixed a real MLX 0.32.0 buffer-reuse hazard: repeated model construction/computation within one process could silently corrupt an unrelated later computation. Root-caused to the buffer-cache pool; fixed with an eval barrier in `mixing_matrices` plus `mx.set_cache_limit(0)`, scoped to the export script and the new test file (not the training/inference hot path).

## Testing
- [x] Tests added/updated (`tests/test_mlx_mixing_matrix.py`, extended `tests/test_parity_fixture_export.py`)
- [x] All tests passing — full `.venv/bin/python -m pytest -q -rs` (1586 passed, 11 skipped), the smoke gate (`scripts/smoke_test.py`), `swift build` + `swift run monica-selfcheck` for both `swift/` and `swift/engine/`, `swift package show-dependencies` empty for `swift/`, `git diff main -- swift/Package.swift` empty
- [ ] Manual testing completed — mlx-swift cannot execute on this host (no Xcode metallib); `swift model parity (macOS, mlx-swift)` CI is authoritative for the new Swift-side gates

### CODE REVIEW

File: scripts/export_parity_fixture.py [hunk 1]
```diff
@@ -116,6 +116,19 @@ def build_fixture(config_path: str, out_dir: str, *, batch: int, seq: int,
...
+    # allocate/free/reuse buffers of the same size class in one process, which this
+    # exporter does far more of than any other caller (repeated regeneration, the #166
+    # staleness test re-invoking `build_fixture` inside an already-populated pytest
+    # process). `mx.set_cache_limit(0)` (proven in isolation to take the observed failure
+    # rate from ~50-100% down to 0/10 trials) trades allocator throughput — irrelevant for
+    # a one-shot fixture export — for correctness; it must not be ported into a hot
+    # training/inference path.
+    mx.clear_cache()
+    mx.set_cache_limit(0)
     mx.random.seed(seed)
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply scripts/export_parity_fixture.py]: ✅ Fixed in 9e87a07ace2dffa78a43fe159a3ac24448816c7c — build_fixture() now captures the prior cache limit via `_prev_cache_limit = mx.set_cache_limit(0)` and restores it in a `finally` that wraps the entire rest of the function (every early `raise SystemExit` refusal path included), so the disable-cache mitigation still covers build_fixture's full duration but no longer leaks into the rest of the pytest process.

File: swift/engine/Sources/MonicaEngine/MonicaModel.swift [hunk 2]
```diff
@@ -141,4 +141,63 @@ public final class MonicaModel: Module {
...
+    /// instead of via a broadcast error deep inside `step`.
+    public func verifyBlock(_ tokens: [Int], _ state: [LayerState])
+        throws -> (logits: [MLXArray], states: [[LayerState]])
+    {
+        for st in state {
+            if let leaf = st.arrays.first, leaf.dim(0) != 1 {
+                throw EngineError.stateMismatch(
+                    "verifyBlock: state batch must be 1, got \(leaf.dim(0))")
+            }
+        }
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply swift/engine/Sources/MonicaEngine/MonicaModel.swift:189]: ✅ Fixed in 9e87a07ace2dffa78a43fe159a3ac24448816c7c — verifyBlock now iterates every leaf of `st.arrays` (conv+ssm / k+v) instead of only `st.arrays.first`, so a batch mismatch confined to the second leaf can no longer slip past the guard.

**Comment by travisgalloway** [comment]
Addressed review feedback:

- @copilot-pull-request-reviewer: both line comments from the review fixed in 9e87a07 — `build_fixture()` now restores the prior MLX cache limit in a `finally` (while keeping the cache disabled for the full duration of the export, preserving the #264 buffer-reuse mitigation), and `verifyBlock`'s batch-1 guard now checks every state leaf instead of only the first.
